File: Shared/src/main/java/com/lenta/shared/models/core/BarcodeInfo.java
package com.lenta.shared.models.core;

import lombok.Getter;

public class BarcodeInfo {
    @Getter private boolean isWeight;
    @Getter private double weight;
    @Getter private String barcode;
    @Getter private String barcodeUom;
    @Getter private String productMaterial;

    public BarcodeInfo(String barcode) {
        this (barcode, null, null, false, 0);
    }

    public BarcodeInfo(String barcode, boolean isWeight, double weight) {
        this(barcode, null, null, isWeight, weight);
    }

    public BarcodeInfo(String barcode, String productMaterial, String barcodeUom, boolean isWeight, double weight)
    {
        this.barcode = barcode;
        this.productMaterial = productMaterial;
        this.barcodeUom = barcodeUom;
        this.isWeight = isWeight;
        this.weight = weight;
    }
}

File: Shared/src/main/java/com/lenta/shared/models/core/BarcodeInfo.kt
package com.lenta.shared.models.core

class BarcodeInfo(val barcode: String, val productMaterial: String?, val barcodeUom: String?, val isWeight: Boolean, val weight: String) {

    constructor(barcode: String) : this(barcode, null, null, false, "0")

    constructor(barcode: String, isWeight: Boolean, weight: String) : this(barcode, null, null, isWeight, weight)

}